### PR TITLE
remove data-default-class

### PR DESCRIPTION
**Pull request by ocramz**
* remove dependency
* define and export `httpConfigDefault` and `retryPolicyDefault` values
* bump bugfix version level

### CODE REVIEW

**Comment by ocramz** [comment]
@mrkkrp I've made this PR because currently making `req` calls with default values forces one to import this extra library, which is unnecessary (and Default is frankly un-idiomatic). I'd like `req` to be self-contained. Hope you like this change!

File: Network/HTTP/Req.hs [hunk 1]
```diff
@@ -638,6 +641,9 @@ instance Default HttpConfig where
     where
       statusCode = Y.statusCode . L.responseStatus
 
+retryPolicyDefault :: forall m . Monad m => RetryPolicyM m
```
[the remark this answers is not in the document]
Comment (ocramz) [reply Network/HTTP/Req.hs]: True, I've recently made a PR to `retry` which exposes this :  https://github.com/Soostone/retry/pull/60/files#diff-fcb684316144775fd2a817cce6b64f60R153

**Comment by ocramz** [comment]
The motivation behind this PR is to avoid getting bitten in the a** about a missing dependency every time one tries to make an HTTP call in a new project. With this "breaking" change we'd go towards a self-contained HTTP library.

**Comment by ocramz** [comment]
@mrkkrp I would love to understand why you think this is a negative change, before closing this. Perhaps I can add a deprecation annotation to the Default instance, which could guide library authors towards the next iteration?

**Comment by mrkkrp** [comment]
Hi @ocramz I'll think about this a bit more. I wasn't online for several days because of my move to Paris (I still only have internet at work and only could login into GitHub today).

**Comment by ocramz** [comment]
Hi @mrkkrp , thank you for the reply! Sure, take your time with the decision and I'd love to help out in moving this forward if needed.

**Comment by mrkkrp** [comment]
I think dropping `data-default` is a good idea. Probably need to do it in all my projects. I'll see this weekend about this particular PR.

**Comment by mrkkrp** [comment]
OK, to proceed with this one I need newer version of `retry` with `data-default` dropped.

**Comment by mrkkrp** [comment]
@ocramz This is blocked on `retry` now. FYI, I dropped `data-default-class` from everywhere, only Req remains now.

**Comment by ocramz** [comment]
@mrkkrp Thank you for this! Perhaps I could ask @MichaelXavier to cut a new release of `retry`.

**Pull request merged**